pkgs/development/tools/build-managers/gradle/default.nix

```diff
@@ -179,4 +180,45 @@ rec {
     hash = "sha256-PiQCKFON6fGHcqV06ZoLqVnoPW7zUQFDgazZYxeBOJo=";
     defaultJava = jdk11;
   };
+
+  wrapGradle = {
+      lib, callPackage, mitm-cache, substituteAll, symlinkJoin, concatTextFile, makeSetupHook,
+    }:
+    gradle-unwrapped:
+    lib.makeOverridable (args:
+    let
+      gradle = gradle-unwrapped.override args;
+      build-support = ../../../../build-support/gradle;
```

Comment:
Relative paths aren't guaranteed to be stable, this needs to be handled better.

Reply:
I'm now using `pkgs.path` for this

Reply:
That would not be any better as it'd still be an arbitrary path that is not stable.

A proper solution would be passing the two files as Nix values.

Reply:
where do you pass them from? all-packages.nix?

Reply:
I don't know the situation but I'd expect them to be passthru attrs of gradle or something.

Reply:
the setup hook files are separate from the Gradle derivation, so where would you even passthrough them

Reply:
I see.

Why though? It'd probably be simpler for everyone if it just lived where gradle lives (or a sub-directory thereof). The file structure in gradle's own dir is stable.

Reply:
that's true, there's no particular reason for that, i changed it to be in the same dir as gradle

Reply:
That's a lot better, thanks!